simtools/corsika/corsika_output_visualize.py

```diff
@@ -29,65 +30,64 @@ def _kernel_plot_2D_photons(corsika_output_instance, property_name, log_z=False)
     -------
     list
         List of figures for the given telescopes.
+    list
+        List of the figure names.
 
     Raises
     ------
     ValueError
-        if `name` is not allowed.
-
-    """
-    x_label = {
-        "counts": "x (m)",
-        "density": "x (m)",
-        "direction": "cos(x)",
-        "time_altitude": "Time since 1st interaction (ns)",
-        "num_photons_per_telescope": "Event number",
-    }
-    y_label = {
-        "counts": "y (m)",
-        "density": "y (m)",
-        "direction": "cos(y)",
-        "time_altitude": "Altitude of emission (km)",
-        "num_photons_per_telescope": "Telescope index",
-    }
-    if property_name not in x_label:
-        msg = f"property_name must be one of {list(x_label.keys())}"
+        if `property` is not allowed.
+    """
+    if property_name not in corsika_output_instance._dict_2D_distributions:
+        msg = f"This property does not exist. The valid entries are " \
+              f"{corsika_output_instance._dict_2D_distributions}"
         _logger.error(msg)
-        raise ValueError(msg)
-
-    if property_name == "counts":
-        hist_values, x_edges, y_edges = corsika_output_instance.get_2D_photon_position_distr(
-            density=False
-        )
-    elif property_name == "density":
-        hist_values, x_edges, y_edges = corsika_output_instance.get_2D_photon_position_distr(
-            density=True
-        )
-    elif property_name == "direction":
-        hist_values, x_edges, y_edges = corsika_output_instance.get_2D_photon_direction_distr()
-    elif property_name == "time_altitude":
-        hist_values, x_edges, y_edges = corsika_output_instance.get_2D_photon_time_altitude()
-    elif property_name == "num_photons_per_telescope":
-        hist_values, x_edges, y_edges = corsika_output_instance.get_2D_num_photons_distr()
-        hist_values, x_edges, y_edges = [hist_values], [x_edges], [y_edges]
+        raise ValueError
+    function = getattr(
+        corsika_output_instance,
+        corsika_output_instance._dict_2D_distributions[property_name]["function"],
+    )
+    hist_values, x_edges, y_edges = function()
 
     all_figs = []
+    fig_names = []
     for i_hist, _ in enumerate(x_edges):
         fig, ax = plt.subplots()
         if log_z is True:
             norm = colors.LogNorm(vmin=1, vmax=np.amax([np.amax(hist_values[i_hist]), 2]))
         else:
             norm = None
         mesh = ax.pcolormesh(x_edges[i_hist], y_edges[i_hist], hist_values[i_hist], norm=norm)
-        ax.set_xlabel(x_label[property_name])
-        ax.set_ylabel(y_label[property_name])
+        if (
+                corsika_output_instance._dict_2D_distributions[property_name]["x edges unit"] is \
```

Comment:
The `\` is not necessary here inside brackets (and shouldn't be used anyway in simtools)